system/jlib/jsocket.cpp

```diff
@@ -1531,7 +1546,17 @@ void CSocket::logPollError(unsigned revents, const char *rwstr)
     if (revents & POLLERR)
     {
         char lname[256];
-        int lport = name(lname, sizeof(lname));
+        int lport;
+        try
+        {
+            lport = name(lname, sizeof(lname));
+        }
+        catch (IException *e)
+        {
+            strcpy(lname, "(NULL)");
+            lport = 1;
+            e->Release();
+        }
```

Comment:
logConnectionInfo may be ok as its always after a successful connect and if there is an error then throwing may make sense.  I will wrap it in try / catch and ignore error if you think that is better.
But in general case - should name() throw ?  I am not sure.